Lib/tkinter/__init__.py

```diff
@@ -1385,11 +1385,14 @@ def bind(self, sequence=None, func=None, add=None):
         return self._bind(('bind', self._w), sequence, func, add)
 
     def unbind(self, sequence, funcid=None):
-        """Unbind for this widget for event SEQUENCE  the
-        function identified with FUNCID."""
-        self.tk.call('bind', self._w, sequence, '')
+        """Unbind for this widget the event SEQUENCE. if
+        FUNCID is given, delete the command also."""
+        bound = ''
         if funcid:
             self.deletecommand(funcid)
```

Comment:
The test failure is here, `_tkinter.TclError: can't delete Tcl command`.  In the absence of more specific information, I interpret this as tk, on Linux, refusing to delete a command that is in use.  Try moving the deletion after the rebinding, as it originally was.  It is OK to have two `if funcid` clauses.

Reply:
Should this work? I just give it a try with ubuntu tests and it worked.
```
    def unbind(self, sequence, funcid=None):
        """Unbind for this widget the event SEQUENCE. if
        FUNCID is given, delete the command also."""
        funcs = self.tk.call('bind', self._w, sequence, '')
        if funcid:
            self.deletecommand(funcid)
            keep = '\n'.join(f for f in funcs.split('\n') if not f.startswith(f'if {{"[{funcid}'))
            self.tk.call('bind', self._w, sequence, keep)
```
Where should the test be written? Would the test just check that after `unbind` with given `funcid`, other bound function are there and the given one is gone?

Reply:
Hey @terryjreedy r you still there? I've done everything you asked except for the new test case, for which I asked you about. As you see the code now passes all tests and could be merged. I mark this as resolved.

Reply:
The Travis test still fails and the revision is not what I asked for, apparently not clearly enough.  A second problem is that the new docstring is what I suggested on the issue better described the current behavior, not the desired behavior.  I have revised the docstring to what I believe Serhiy wanted and the the code to match.  If CI passes, we can consider a new test.